tests/plugins/data.greylist.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+var stub         = require('../fixtures/stub');
+var Plugin       = require('../fixtures/stub_plugin');
+var Connection   = require('../fixtures/stub_connection');
+var Address      = require('../../address').Address;
+var config       = require('../../config');
+var ResultStore  = require('../../result_store');
+
+var _set_up = function (done) {
+
+    this.plugin = new Plugin('data.greylist');
+    this.plugin.config = config;
+    this.plugin.register();
+
+    this.connection = Connection.createConnection();
+    this.connection.results = new ResultStore(this.connection);
+    this.connection.transaction = {
+        results: new ResultStore(this.connection),
+    };
+
+    done();
+};
+
+/* jshint maxlen: 100 */
+exports.in_list = {
+    setUp : _set_up,
+    'is_sender_special, postmaster@example.com': function (test) {
+        test.expect(1);
+        test.equal(true,  this.plugin.is_sender_special('postmaster@example.com'));
+        test.done();
+    },
+    'is_sender_special, double-bounce@example.com': function (test) {
+        test.expect(1);
+        test.equal(true,  this.plugin.is_sender_special('double-bounce@example.com'));
+        test.done();
+    },
+    'is_sender_special, double-bounce-12345678@example.com': function (test) {
+        test.expect(1);
+        test.equal(true,  this.plugin.is_sender_special('double-bounce-12345678@example.com'));
+        test.done();
+    },
+    '!is_sender_special, double-bounce-123456@example.com': function (test) {
+        test.expect(1);
+        test.equal(true,  this.plugin.is_sender_special('double-bounce-123456@example.com'));
+        test.done();
+    },
+    '!is_sender_special, postwaster@example.com': function (test) {
+        test.expect(1);
+        test.equal(true,  this.plugin.is_sender_special('postwaster@example.com'));
+        test.done();
+    },
+    'white, rcpt': function (test) {
+        this.plugin.whitelist = {
+            "mail":{"josef@examlpe.com":true},
+            "rcpt":{"josef@examlpe.net":true},
+            "ip":[
+                [{"octets":[123,123,123,234]},32],
+                [{"parts":[10754,33284,54784,32864,31008,16448,8430,38528]},128],
+                [{"octets":[123,210,123,234]},27],
+                [{"parts":[10754,33284,54784,32864,31008,61166,0,0]},96]
+            ]
+        };
+        this.plugin.list = {"dyndom":["sgvps.net"]};
+        test.expect(7);
+        test.equal(true,  this.plugin.addr_in_list('mail', 'josef@examlpe.com'));
```

Comment:
Ahah, I think I see your problem.

````
this.plugin.whitelist = {
+            "mail":{"josef@examlpe.com":true},
````

You are setting this.plugin.whitelist.mail above.

Here you are accessing this.plugin.mail.